### PR TITLE
Remove consrc for PostgreSQL 12 compat

### PR DESCRIPTION
**Pull request by andrewclink**
Removes `consrc` for check constraints by transforming the result of `pg_get_constraintdef`. All tests passing against PostgreSQL 12.1.

### CODE REVIEW

**Comment by andrewclink** [comment]
I tried squashing my commits but screwed something up. Unfortunately I have other deadlines. If someone wants to help me learn how to correct this I'd very much appreciate it.

File: spec/dummy/db/schema.rb [hunk 1]
```diff
@@ -2,11 +2,11 @@
 # of editing this file, please use the migrations feature of Active Record to
 # incrementally modify your database, and then regenerate this schema definition.
 #
-# Note that this schema.rb definition is the authoritative source for your
-# database schema. If you need to create the application database on another
-# system, you should be using db:schema:load, not running all the migrations
-# from scratch. The latter is a flawed and unsustainable approach (the more migrations
-# you'll amass, the slower it'll run and the greater likelihood for issues).
+# This file is the source Rails uses to define your schema when running `rails
```
Comment (betesh) [inline spec/dummy/db/schema.rb]: Don't check in the changes in this file -- If I run tests with an older version of Rails, it will revert this file back to the old version.  If you want to ignore it, untrack changes locally (google `git --assume-unchanged` for how to do this).

File: lib/active_record/postgres/constraints/schema_dumper.rb [hunk 2]
```diff
@@ -16,6 +16,7 @@ def indexes_in_create(table, stream)
         private
 
         def dump_indexes(indexes, stream)
+          puts "Dumping indexes: #{indexes.inspect}"
```
Comment (betesh) [inline lib/active_record/postgres/constraints/schema_dumper.rb]: Revert the change in this file--we don't want this output in the released gem.

File: lib/active_record/postgres/constraints/types/unique.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module ActiveRecord
```
Comment (betesh) [inline lib/active_record/postgres/constraints/types/unique.rb]: This class duplicates a lot of code in check.rb.  Move that code into a shared module instead of duplicating it.

Also please add tests.

**Comment by andrewclink** [comment]
Sorry, the last few weren't intended to be part of this PR. Should be fixed now.

**Comment by betesh** [comment]
Looks like all the tests are passing now (2 failed but I restarted them -- unfortunately, there are intermittent timing failures that I have tried very hard to fix but so far have been unsuccessful).  Thank you for the quick turnaround.

However, please see my comment here: https://github.com/on-site/active_record-postgres-constraints/issues/21#issuecomment-560539945 -- I am concerned that just because this works with one version of pg does not guarantee us it will remain backwards compatible, so we need to test with multiple versions of pg on travis.

**Comment by andrewclink** [comment]
For what it's worth, the tests also pass against pg 11. I don't know when `pg_get_constraintdef()` was introduced, possibly 9.5.

**Comment by betesh** [comment]
True, but even without this change, tests were passing on travis, which means we weren't testing enough scenarios.

**Comment by betesh** [comment]
I merged #25 into master.  Please rebase this onto master and ensure tests pass.

**Comment by ziaulrehman40** [comment]
Lets rebase and close this please.

**Comment by betesh** [comment]
Fixed in [0.2.1](https://rubygems.org/gems/active_record-postgres-constraints/versions/0.2.1)
